### PR TITLE
Closes #17531: Use top sites count fact for top sites telemetry

### LINKED ISSUE #17531: [Bug] top sites metric remain set to false
Followup to #16832

## Prerequisites
Have the default top sites available

## Steps to reproduce
Generate the **`metrics`** ping

### Expected behavior
The following **`boolean`** probes should be properly updated:
```
"metrics": {
        "boolean": {
          ...
          "metrics.has_top_sites"
         ...
        }
```

### Actual behavior
The above mentioned probes remain set to false

## Notes
Similar to #17530, this metric is [set on startup](https://github.com/mozilla-mobile/fenix/blob/cf61c10c39be1f7e2037a36408577e1dc57667d8/app/src/main/java/org/mozilla/fenix/components/metrics/GleanMetricsService.kt#L756) for Glean. This metric  should **always reflect the current state of the app**. Currently, we are only evaluating this metric (and others) on startup and this can lead to delayed/incorrect metrics.

### PR DESCRIPTION
**Pull request by rocketsroger**
### Pull Request checklist

- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/test/java/org/mozilla/fenix/components/metrics/MetricControllerTest.kt [hunk 1]
```diff
@@ -167,4 +172,41 @@ class MetricControllerTest {
...
+        assertEquals(controller.factToEvent(fact), Event.HaveNoTopSites)
+
+        fact = Fact(
+            Component.FEATURE_TOP_SITES,
+            Action.INTERACTION,
+            TopSitesFacts.Items.COUNT,
+            "10"
+        )
+
+        assertEquals(controller.factToEvent(fact), Event.HaveTopSites)
```
Comment (gabrielluong) [inline app/src/test/java/org/mozilla/fenix/components/metrics/MetricControllerTest.kt:210]: Looks good, maybe we should just add the case where the count is a negative integer (-4) as well

**Review by gabrielluong — APPROVED** [review]

**Comment by gabrielluong** [comment]
Ah, we should remove this some parts of this as well https://github.com/mozilla-mobile/fenix/blob/eaef276345d018d8de6c2eff3ac0b00ecf0cbca2/app/src/main/java/org/mozilla/fenix/components/metrics/GleanMetricsService.kt#L812-816

**Comment by rocketsroger** [comment]
> Ah, we should remove this some parts of this as well https://github.com/mozilla-mobile/fenix/blob/eaef276345d018d8de6c2eff3ac0b00ecf0cbca2/app/src/main/java/org/mozilla/fenix/components/metrics/GleanMetricsService.kt#L812-816

Unfortunately during my investigation we can't remove those just yet.  The start up telemetry still depends on these stored values.

File: app/src/test/java/org/mozilla/fenix/components/metrics/MetricControllerTest.kt (hunk 1 above)
Comment (rocketsroger) [reply app/src/test/java/org/mozilla/fenix/components/metrics/MetricControllerTest.kt:210]: Agreed.  I'll add a test case for negative count.  Thanks,
